src/twin/mod.rs

```diff
@@ -174,6 +175,7 @@ impl Twin {
             web_service,
             tx_reported_properties,
             tx_outgoing_message,
+            authenticated: false,
             authenticated_once: false,
```

Comment:
I'm not 100% sure how the protocol flow is here: what happens after initial authentication when we set `authenticated` to false again? Do we have to reset `authenticated_once` to false, as well? IMHO it would be then more beneficial to introduce sth. like:

```rust
enum AuthenticationState {
    UNAUTHENTICATED,
    INITIALLY_AUTHENTICATED,
    AUTHENTICATED,
}
```